OpenRA.Game/Network/Order.cs

```diff
@@ -227,19 +271,26 @@ public byte[] Serialize()
 						w.Write(UIntFromActor(Subject));
 
 						OrderFields fields = 0;
-						if (TargetActor != null) fields |= OrderFields.TargetActor;
-						if (TargetLocation != CPos.Zero) fields |= OrderFields.TargetLocation;
+						if (Target.SerializableType != TargetType.Invalid) fields |= OrderFields.Target;
 						if (TargetString != null) fields |= OrderFields.TargetString;
 						if (Queued) fields |= OrderFields.Queued;
 						if (ExtraLocation != CPos.Zero) fields |= OrderFields.ExtraLocation;
 						if (ExtraData != 0) fields |= OrderFields.ExtraData;
 
 						w.Write((byte)fields);
 
-						if (TargetActor != null)
-							w.Write(UIntFromActor(TargetActor));
-						if (TargetLocation != CPos.Zero)
-							w.Write(TargetLocation);
+						if (Target.SerializableType != TargetType.Invalid)
+							w.Write((byte)Target.Type);
+						if (Target.SerializableType == TargetType.Actor)
+							w.Write(UIntFromActor(Target.SerializableActor));
+						if (Target.SerializableType == TargetType.FrozenActor)
+						{
+							w.Write(Target.FrozenActor.Owner.InternalName);
```

Comment:
Is there a reason you are using ```InternalName``` instead of the PlayerActor's `ActorID`?

Reply:
Because i'm a doofus.  Good idea.